backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/ContextInitializer.scala

```diff
@@ -138,3 +142,14 @@ class ContextInitializer extends ContextApi {
     // TODO shutdown implementation in velox to release resources
   }
 }
+
+object ContextInitializer {
+  var init: Boolean = false
+  def initializeNative(conf: Map[String, String]): Unit = {
+    if (init) {
+      return
+    }
+    NativeBackendInitialized.initializeBackend(conf)
+    init = true
+  }
```

Comment:
Looks like something needs to be guarded by a lock?

Reply:
BTW `AtomicBoolean` would be preferred if we want to avoid using non-final static variables.

Reply:
It's safe here and impossible for multi thread invoke ContextInitializer#initializeNative.

Reply:
I think it's not enough to rely on outside code (e.g. from vanilla Spark) to guarantee the thread-safety here. Especially when driver and executor can be launched in a single JVM process in Spark's local mode. Would you please double-check?

Reply:
I modify these codes and verified when running UT, if run `NativeBackendInitialized.initializeBackend` twice in single JVM, it will core then exit. CI has prove its safety right?

Reply:
> I modify these codes and verified when running UT, if run NativeBackendInitialized.initializeBackend twice in single JVM, it will core then exit. CI has prove its safety right?

Yes it probably can be considered safe in CI but I am not sure CI covers all cases. If it's hard to verify all the cases, I think it's OK to just add some guards to the code.

Reply:
Or we can check vanilla Spark's code for its local mode. If the calls to the method strictly happens in the same thread, we can leave some comment here to explain why the code doesn't need to be guarded by lock.

Reply:
There are only two cases: local or cluster, let me double check the initialization of local mode.

Reply:
![image](https://github.com/oap-project/gluten/assets/34979747/1fe53b75-dc4c-41fa-a393-954082578721)
![image](https://github.com/oap-project/gluten/assets/34979747/0253f84b-a54a-48f3-98bb-28ec80d82ff1)

Verified and add comments.

Reply:
OK. Thanks for making the change.